### DISCUSSION TITLE
Deploying across multiple panels

### DISCUSSION THREAD

**Discussion by rudgej** [Blackymas/NSPanel_HA_Blueprint#1690] [Q&A]
I have 15 panels. 

They were mostly "ok" in terms of what they did but I have spent the last 2 days creating the "perfect" panel with chips, sensors, entities and button pages 3 and 4 with standard house control items. 

So now I want to add in button pages 1 and 2 from the exiting panels and deploy it to each panel. 

So I assume I could just cost paste some yaml.

BUT ...

The order of YAML seems reasonably random - it does not flow button page, entites, Chips etc - but has some degree of randomness where part of a chip will be in a different part of the code.

I tried splitting the code up and re organising it - but that just got undone by the system when you save. Likewise comments ( plenty of HA feature requests have said this is not possible).

So there now seem 2 options 

1. Duplicate the " perfect" panel and then manually add in pages 1 and 2

2. Duplicate the panel and do a hack of cut paste YAML as best I can 

Both are time consuming.

There is then the other issue of maintaining this as any change needs to ( ideally ) be cascaded over all 15 panels.

It is not quite as bad as it seems as I have 2 panels in the kitchen , 3 in one bedroom and 4 in the halls, so there are "only" 9 unique panel formate but still.....

Does anyone have any better suggestions as to how this can best be managed ?

What I really wanted to do was just paste in my template and then just paste in button Page 1 and 2 but this seem impractical 

Thank you

Jeremy

**Comment by rudgej** [comment]
Anyone!

This is just driving me nuts

The YAML is basically random - nothing is held together so when I cut past the screen just turns red as it is invalid. 

I then have to try and find out why ( which is mostly unsuccessful).

As an example a YAML snippet is below.  As just one example Entity 17 is not  together but split. There is no logical reason for this. Likewise 19 etc.

In this example they are close but in the whole code entities can be random all over the place.

Is there any way to get it to produce logical code that flows? 

I am not spring longer to try ( and fail) to deploy it then it took to set up!




   entities_entity14: light.group_bedroom
    entities_entity14_name: Bedroom
    entities_entity15: switch.group_external
    entities_entity15_icon: ""
    entities_entity15_name: Outside
    entities_entity17: sensor.ac_aboe_gge_on
    entities_entity18: sensor.ac_art_room
    entities_entity18_name: Art Room
    entities_entity19: sensor.ac_cinema
    entities_entity19_name: "Cinema "
    entities_entity17_name: "Above Garage "
    entity_page03_label: "AC Status "
    entities_entity17_icon: mdi:air-conditioner
    entities_entity18_icon: mdi:air-conditioner
    entities_entity19_icon: mdi:air-conditioner
    entities_entity20: sensor.ac_guest
    entities_entity20_name: "Guest Room "
    entities_entity20_icon: mdi:air-conditioner
    entities_entity21: sensor.ac_gym
    entities_entity21_name: Gym
    entities_entity21_icon: mdi:air-conditioner
    entities_entity22: sensor.ac_harrison
    entities_entity22_name: "Harrison "
    entities_entity22_icon: mdi:air-conditioner

**Comment by rudgej** [comment]
Bump ?!

**Comment by edwardtfn** [comment]
The yaml is not random. A key is created when you set a value in the Blueprint settings, so the yaml is following the order you set the fields.

You shoukd be able to reorder those in your `automations.yaml` file, then reload automations under Developer Tools.
